backend/tracim_backend/lib/core/storage.py

```diff
@@ -0,0 +1,308 @@
+                    page_number=preview_generator_page_number, file_id=depot_file.file_id
+                )
+            )
+        return preview_generator_page_number
+
+    def _get_valid_content_filepath(
+        self,
+        depot_stored_file: StoredFile,
+        prefix: str = "tracim-undefined",
+        file_extension: str = "",
+    ) -> typing.Generator[int, None, None]:
+        """
+        Generic way to get content filepath for all depot backend.
+        :param depot_stored_file: content as depot StoredFile
+        :param file_extension: extension of the file we expect
+        :return: content filepath
+        """
+
+        file_label = "{prefix}-{file_id}".format(prefix=prefix, file_id=depot_stored_file.file_id)
+        base_path = "{temp_dir}/{file_label}".format(
+            temp_dir=tempfile.gettempdir(), file_label=file_label,
+        )
+
+        file_path = "{base_path}{file_extension}".format(
+            base_path=base_path, file_extension=file_extension,
+        )
+
+        lockfile_path = "{base_path}{file_extension}".format(
+            base_path=base_path, file_extension=".lock",
+        )
+        # FIXME - G.M - 2020-01-05 - This will create a lockfile for
+        # each depot file we do need (each content revision)
+        # This file will NOT be removed at the end on Linux (FileLock use flock):
+        # see  https://stackoverflow.com/questions/17708885/flock-removing-locked-file-without-race-condition
+        # some investigation needs to be conducted to see if another solution is possible avoiding creating
+        # too many files. See https://github.com/tracim/tracim/issues/4014
+        with filelock.FileLock(lockfile_path):
+            try:
+                # HACK - G.M - 2020-01-05 - This mechanism is inefficient because it
+                # generates a temporary file each time
+                # Improvements need to be made in preview_generator itself
+                # to handle more properly these issues.
+                # We do rely on consistent path based on gettemdir(),
+                # normally /tmp to give consistent path, this is a quick fix which does
+                # not need any change in preview-generator.
+                # note: this base path is configurable through an envirnoment var according
+                # to the Python doc:
+                # https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir
+                with open(file_path, "wb",) as tmp:
+                    tmp.write(depot_stored_file.read())
+                    yield file_path
+            finally:
+                try:
+                    os.unlink(file_path)
+                except FileNotFoundError:
+                    pass
+
+    def _get_valid_content_filepath_legacy(
+        self, depot_stored_file: StoredFile
+    ) -> typing.Generator[int, None, None]:
+        """
+        Legacy way to get content filepath, only work for local file backend of depot
```

Comment:
```suggestion
        Optimized way to get content filepath, only work for local file backend of depot
```

Reply:
are we sure we will keep it ?
It has some drawbacks (uncorrect file-extension for examples)

Note: this method is not new, i just moved it, i can rename it there but that not the point of the PR.